portfolio/src/main/webapp/script.js

```diff
@@ -63,21 +63,51 @@ function previousExperience() {
   selectedExperience--;
 }
 
-function getServerComments() {
-  fetch('/data').then(response => response.json()).then((commentSection) => {
+function getNumCommentsSelected(selected) {
+  var selectedValue = selected.value;
+  getServerComments(selectedValue);
+}
+
+function getServerComments(numCommentsSelected) {
+  fetch('/data?number-comments='+numCommentsSelected).then(response => response.json()).then((commentSection) => {
     populateComments(commentSection);
   });
 }
 
 function populateComments(commentSection) {
   const individualComments = document.getElementById('comments-container');
+  individualComments.innerHTML = '';
   commentSection.forEach((element) => {
-      individualComments.appendChild(createComment(element));
+    individualComments.appendChild(createComment(element));
   });
+  if (sessionStorage.getItem('selectedValue') === null){
+    document.getElementById('number-comments').options[1].selected = true;
```

Comment:
Is this necessary? It looks like this option will always be selected when you first load the page, since you've added this to the HTML

Reply:
Since the sessionStorage only comes into play onchange, when you first load the page the default value hadn't been stored yet (at least that's what seemed to be happening, and adding this fixed it). Would it be okay to leave it in?

Reply:
As far as I can tell, what is happening here is that you're manually setting the `<option>` in the UI to be selected, which then triggers an `onchange` call and so on, with the goal of ultimately setting the `selectedValue` in `sessionStorage`.

What I'm saying is that your default `<option>` is already marked as `selected` in the HTML, so you could skip that step altogether and replace that line with `sessionStorage.setItem(....)`, unless I'm missing something here.

Reply:
I see what you're saying, my apologies, that makes sense.